sql/item_create.cc

```diff
@@ -4761,6 +4771,14 @@ Create_func_rand::create_native(THD *thd, LEX_CSTRING *name,
   return func;
 }
 
+Create_func_release_all_locks Create_func_release_all_locks::s_singleton;
+Item*
+Create_func_release_all_locks::create_builder(THD *thd)
+{
+  thd->lex->set_stmt_unsafe(LEX::BINLOG_STMT_UNSAFE_SYSTEM_FUNCTION);
+  thd->lex->safe_to_cache_query= 0;
+  return (new (thd->mem_root) Item_func_release_all_locks(thd));
```

Comment:
Other lock related functions use:
`thd->lex->uncacheable(UNCACHEABLE_SIDEEFFECT);`
Shouldn't this function do the same?

Reply:
If is_free_lock and is_used_lock have (UNCACHEABLE_SIDEEFFECT)
Release_all_locks should for sure :)

Reply:
Redundant parentheses here.